extensions/default/src/validateInstances.ts

```diff
@@ -0,0 +1,319 @@
+import { vec3 } from 'gl-matrix';
+import toNumber from '@ohif/core/src/utils/toNumber';
+import sortInstancesByPosition from '@ohif/core/src/utils/sortInstancesByPosition';
+import {
+  hasPixelMeasurements,
+  hasOrientation,
+  hasPosition,
+  _getPerpendicularDistance,
+  _getSpacingIssue,
+  reconstructionIssues,
+  _isSameOrientation,
+  constructableModalities,
+} from '@ohif/core/src/utils/isDisplaySetReconstructable';
+import { displaySetMessageCodes, displayServiceMessageList } from '@ohif/core';
+
+/**
+ * Calculates the scanAxisNormal based on a image orientation vector extract from a frame
+ * @param {*} imageOrientation
+ * @returns
+ */
+function calculateScanAxisNormal(imageOrientation) {
+  const rowCosineVec = vec3.fromValues(
+    imageOrientation[0],
+    imageOrientation[1],
+    imageOrientation[2]
+  );
+  const colCosineVec = vec3.fromValues(
+    imageOrientation[3],
+    imageOrientation[4],
+    imageOrientation[5]
+  );
+  return vec3.cross(vec3.create(), rowCosineVec, colCosineVec);
+}
+
+/**
+ * Checks if a series is reconstructable to a 3D volume.
+ *
+ * @param {Object[]} instances An array of `OHIFInstanceMetadata` objects.
+ */
+export default function validateInstances(
+  instances: Array<any>,
+  isReconstructable: boolean
+): displayServiceMessageList {
+  const messages = new displayServiceMessageList();
```

Comment:
I don't feel like we should instantiate a new service class from a validation method. If you decided to create a new service to handle messages it should have an instance e.g. ```platform/core/src/services/UINotificationService/index.ts``` and be instantiated and shared along with other OHIF services across the codebase.

Also, the service names should be Uppercased and have simple naming .e.g MessageService. That's why I thought about creating a new manager just for validation in my own sketch (ValidationManager) but it could also be MessageManager or something alike.

Reply:
Itsa not a service. Changed the name to DisplaySetMessageList